### PR TITLE
Devex_1597 - Change the init to create jenkins.service

### PR DESCRIPTION
**Pull request by Galofers**
Modify Jenkins init 
add some changes in service template to create correctly the jenkins service with the new versions
change gpg key in the repo config

## Jira related issue

https://theknotww.atlassian.net/browse/DEVEX-1597

### CODE REVIEW

**Review by aballesta — APPROVED** [review]

File: manifests/repo.pp [hunk 1]
```diff
@@ -3,7 +3,7 @@
 #
 class jenkins::repo(
   Stdlib::Httpurl $base_url = 'https://pkg.jenkins.io',
-  String $gpg_key_filename = 'jenkins.io.key',
+  String $gpg_key_filename = 'jenkins.io-2023.key',
```
Comment (cmartinez85) [inline manifests/repo.pp:6]: is the old jenkins instance affected by this modification?

Comment (Galofers) [reply manifests/repo.pp:6]: Only affects when you install a new Jenkins package.

File: manifests/cli.pp [hunk 2]
```diff
@@ -35,26 +35,27 @@
...
+  #}
   #  ~> exec { 'jenkins-cli' :
   #    command     => "${extract_jar} && ${move_jar} && ${remove_dir}",
   #    path        => ['/bin', '/usr/bin'],
   #    cwd         => '/tmp',
   #    refreshonly => true,
   #  }
   # Extract latest CLI in case package is updated / downgraded
-  Package[$::jenkins::package_name] ~> Exec['jenkins-cli']
+  #Package[$::jenkins::package_name] ~> Exec['jenkins-cli']
```
Comment (cmartinez85) [inline manifests/cli.pp]: Could you remove the commented code?

File: manifests/cli.pp [hunk 3]
```diff
@@ -35,26 +35,27 @@
...
 
   file { $jar:
     ensure  => file,
     mode    => '0644',
-    require => Exec['jenkins-cli'],
+    #require => Exec['jenkins-cli'],
```
Comment (cmartinez85) [inline manifests/cli.pp]: same here

**Review by cmartinez85 — APPROVED** [review]

File: manifests/cli.pp (hunk 3 above)
Comment (Galofers) [reply manifests/cli.pp]: done

**Pull request merged**